hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/txn/DefaultHandleWorkflow.java

```diff
@@ -0,0 +1,98 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.workflows.handle.flow.txn;
+
+import static com.hedera.node.app.state.logging.TransactionStateLogger.logStartUserTransaction;
+import static com.hedera.node.app.state.logging.TransactionStateLogger.logStartUserTransactionPreHandleResultP2;
+import static com.hedera.node.app.state.logging.TransactionStateLogger.logStartUserTransactionPreHandleResultP3;
+
+import com.hedera.node.app.records.BlockRecordManager;
+import com.hedera.node.app.workflows.handle.StakingPeriodTimeHook;
+import com.hedera.node.app.workflows.handle.flow.dispatch.logic.DispatchProcessor;
+import com.hedera.node.app.workflows.handle.flow.txn.logic.HollowAccountCompleter;
+import com.hedera.node.app.workflows.handle.flow.txn.logic.SchedulePurger;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * The default implementation of the handle workflow.
+ */
+@Singleton
+public class DefaultHandleWorkflow {
+    private static final Logger logger = LogManager.getLogger(DefaultHandleWorkflow.class);
+
+    private final StakingPeriodTimeHook stakingPeriodTimeHook;
+    private final BlockRecordManager blockRecordManager;
+    private final SchedulePurger schedulePurger;
+    private final DispatchProcessor dispatchProcessor;
+    private final HollowAccountCompleter hollowAccountFinalization;
+
+    @Inject
+    public DefaultHandleWorkflow(
+            final StakingPeriodTimeHook stakingPeriodTimeHook,
+            final BlockRecordManager blockRecordManager,
+            final SchedulePurger schedulePurger,
+            final DispatchProcessor dispatchProcessor,
+            final HollowAccountCompleter hollowAccountFinalization) {
+        this.stakingPeriodTimeHook = stakingPeriodTimeHook;
+        this.blockRecordManager = blockRecordManager;
+        this.schedulePurger = schedulePurger;
+        this.dispatchProcessor = dispatchProcessor;
+        this.hollowAccountFinalization = hollowAccountFinalization;
+    }
+
+    /**
+     * Executes the handle workflow. This method is the entry point for handling a user transaction.
+     * It processes the staking period time hook, advances the consensus clock, expires schedules, logs the
+     * user transaction,
+     * @param userTxn the user transaction component
+     * @return the work done
+     */
+    public WorkDone execute(UserTransactionComponent userTxn) {
+        processStakingPeriodTimeHook(userTxn);
```

Comment:
Where did publishing genesis records end up? Should that not be here too?

Reply:
It is inside `UseTxnWorkflow` in this same PR